glRemixRenderer/mesh_loader.cpp

```diff
@@ -29,14 +32,57 @@ bool glRemix::load_mesh_from_path(std::filesystem::path asset_path,
         return false;
     }
 
-    // fastgltf::validate(asset.get());
+    // get textures
+    for (UINT i = 0; i < 1; ++i)  // just get base color texture for now
+    {
+        const auto& img = asset->images[i];
+
+        // load image using DirectXTex
+        ScratchImage scratch_image;
+        TexMetadata metadata{};
+
+        if (auto* uri = std::get_if<fastgltf::sources::URI>(&img.data))
+        {
+            std::filesystem::path uri_path = uri->uri.fspath();
+            std::filesystem::path path = asset_path.parent_path() / uri_path;
+
+            HRESULT hr = LoadFromWICFile(path.c_str(), WIC_FLAGS_FORCE_RGB, &metadata,
+                                         scratch_image);
+
+            if (FAILED(hr))
+            {
+                printf("Failed to load texture file: %s\n", path.string().c_str());
+                return false;
+            }
+        }
+
+        // assume that scratch_image is the correct format
+        const Image* img_data = scratch_image.GetImage(0, 0, 0);
+        const size_t byte_size = img_data->slicePitch;
+        auto pixel_buffer = std::make_unique<uint8_t[]>(byte_size);
+        memcpy(pixel_buffer.get(), img_data->pixels, byte_size);
+
+        out_texture.index = static_cast<UINT32>(i);
+        out_texture.desc = {
+            static_cast<UINT32>(img_data->width), static_cast<UINT32>(img_data->height), 1,    1,
+            DXGI_FORMAT_R8G8B8A8_UNORM,           D3D12_RESOURCE_DIMENSION_TEXTURE2D,    false
+        };
+        out_texture.pixels.assign(pixel_buffer.get(), pixel_buffer.get() + byte_size);
```

Comment:
Just assign `out_texture.pixels` directly
```suggestion
        out_texture.index = i;
        out_texture.desc = {
            static_cast<UINT32>(img_data->width), static_cast<UINT32>(img_data->height), 1,    1,
            metadata.format,           D3D12_RESOURCE_DIMENSION_TEXTURE2D,    false
        };
        out_texture.pixels.assign(scratch_image.GetPixels(), scratch_image.GetPixels() + bytes);
```
where bytes uses the metadata size, format is also directly from the texture, otherwise the descriptor is not going to be made correctly.